### PR TITLE
Patch tokio to vendored version

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
This issue still exists for RPC operators: https://github.com/solana-labs/solana/issues/24644

#### Summary of Changes
Patch tokio to `solana-tokio` at v1.29.1 with 4eed411 manually reverted

### CODE REVIEW

**Comment by steviez** [comment]
@CriesofCarrots - Moving some discussion from Discord to here, do you think we should "finish the revert" in light of you finding that piece of the commit that I missed? The revert seemed to help as-is, but I still feel like we should fix the mistake.

Assuming you agree with fixing that, I'm wondering if we should fix up the revert commit first, before moving to vendored tokio

File: Cargo.toml [hunk 1]
```diff
@@ -537,3 +538,21 @@ rev = "6105d7a5591aefa646a95d12b5e8d3f55a9214ef"
...
+
+# Solana RPC nodes experience stalls when running with `tokio` containing this
+# commit:
+# https://github.com/tokio-rs/tokio/commit/4eed411519783ef6f58cbf74f886f91142b5cfa6
+#
+# Tokio maintainers believe performance degradation is due to application bugs:
+# https://github.com/tokio-rs/tokio/issues/4873#issuecomment-1198277677
+#
+# This may indeed be true of the code in this monorepo, but we haven't yet
+# identified the big or a way to fix. As a stopgap, this patches `tokio` to the
```
Comment (steviez) [inline Cargo.toml:550]: nit: "big" ==> "bug"

**Review by steviez — COMMENTED** [review]
Just the one nit on fixing typo in comment; otherwise, these looks good to me. Thanks for writing the detailed comment as well

**Review by steviez — APPROVED** [review]
Looks good to me, but maybe wouldn't hurt to get t-nelson's blessing as well

**Review by t-nelson — APPROVED** [review]
good comment! :shipit:

**Comment by CriesofCarrots** [comment]
I took some extra time to confirm that this would not break downstream users pulling in solana client crates, the way that pinning the tokio version did. All built as expected.

**Pull request merged**